src/lib/Tab/LocationView/LocationsTab.php

```diff
@@ -112,12 +143,27 @@ public function getTemplateParameters(array $contextParameters = []): array
         $versionInfo = $content->getVersionInfo();
         $contentInfo = $versionInfo->getContentInfo();
         $locations = [];
+        $pagination = null;
 
         if ($contentInfo->published) {
-            $locationsDataset = $this->datasetFactory->locations();
-            $locationsDataset->load($contentInfo);
+            $currentPage = $this->requestStack->getCurrentRequest()->query->getInt(
+                self::PAGINATION_PARAM_NAME, 1
+            );
+
+            $locationQuery = new LocationQuery([
+                'filter' => new Query\Criterion\ContentId($contentInfo->id),
+            ]);
+
+            $pagination = new Pagerfanta(
+                new LocationSearchAdapter(
+                    $locationQuery,
+                    $this->searchService
+                )
+            );
 
-            $locations = $locationsDataset->getLocations();
+            $pagination->setMaxPerPage($this->defaultPaginationLimit);
+            $pagination->setCurrentPage(min(max($currentPage, 1), $pagination->getNbPages()));
```

Comment:
From my personal perspective, if user requests a page out of bounds, we should not try to correct him and force the last page (so no `min` call, and no `$pagination->getNbPages()`). Just an opinion thought.

Reply:
I agree and if I recall correctly we validate requested page in other places where pagination is used.